setup.go

```diff
@@ -0,0 +1,104 @@
+package ocp_dnsnameresolver
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/coredns/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+)
+
+const pluginName = "ocp_dnsnameresolver"
+
+var log = clog.NewWithPlugin(pluginName)
+
+func init() { plugin.Register(pluginName, setup) }
+
+func setup(c *caddy.Controller) error {
+	resolver, err := resolverParse(c)
+	if err != nil {
+		return plugin.Error(pluginName, err)
+	}
+
+	onStart, onShut, err := resolver.initPlugin()
+	if err != nil {
+		return plugin.Error(pluginName, err)
+	}
+
+	if onStart != nil {
+		c.OnStartup(onStart)
+	}
+	if onShut != nil {
+		c.OnShutdown(onShut)
+	}
```

Comment:
`initPlugin` always returns valid startup and shutdown functions if there is no error. The check for `nil` is redundant.